exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java

```diff
@@ -47,16 +47,25 @@
   private List<FileStatus> statuses;
 
   public List<String> files;
+  /**
+   * root path for the selections
+   */
   public final String selectionRoot;
+  /**
+   * root path for the metadata cache file (if any)
+   */
+  public final String cacheFileRoot;
```

Comment:
When singlePartitionOpt is applied,  is it possible to update selectionRoot to be cacheFileRoot? That is, we do not maintain cacheFileRoot separately. In stead, a FileSelection with updated selectionRoot is used when singlePartitionOpt is applied.